### PR TITLE
[ADD] document_hosting

### PR DESCRIPTION
**Pull request by elouan-lb**
Previously module `easy_my_coop_document` in `easy_my_coop` repository. See https://github.com/coopiteasy/vertical-cooperative/pull/40

### CODE REVIEW

File: document_hosting/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    'description': """
+    """,
+
+    'author': 'Rémy Taymans',
+    'license': 'AGPL-3',
+    'version': '12.0.1.0.0',
+    'website': "https://github.com/coopiteasy/addons",
+
+    'category': 'Cooperative Management',
+
```
Comment (houssine78) [inline document_hosting/__manifest__.py]: It's more a generic module now. Document is more accurate

File: document_hosting/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+{
+    'name': 'Document Hosting',
+
+    'summary': """
+    Manage the documents of your cooperative.
+    """,
+    'description': """
+    """,
+
+    'author': 'Rémy Taymans',
```
Comment (houssine78) [inline document_hosting/__manifest__.py]:
```suggestion
   "author": "Coop IT Easy SCRLfs",
```

File: document_hosting/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+
+# Copyright 2018 Rémy Taymans <remytaymans@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
Comment (houssine78) [inline document_hosting/__manifest__.py]:
```suggestion
# Copyright 2018 - Today Coop IT Easy SCRLfs (<http://www.coopiteasy.be>)
# - Rémy Taymans <remy@coopiteasy.be>
# - Elouan Le bars <elouan@coopiteasy.be>
# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
```

File: document_hosting/views/document_hosting_menu.xml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+     Copyright 2018 Rémy Taymans <remytaymans@gmail.com>
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
Comment (houssine78) [inline document_hosting/views/document_hosting_menu.xml]:
```suggestion
# Copyright 2018 - Today Coop IT Easy SCRLfs (<http://www.coopiteasy.be>)
```

File: document_hosting/views/document_hosting_menu.xml [hunk 5]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+     Copyright 2018 Rémy Taymans <remytaymans@gmail.com>
```
Comment (houssine78) [inline document_hosting/views/document_hosting_menu.xml]:
```suggestion
     Copyright 2018 - Today Coop IT Easy SCRLfs (<http://www.coopiteasy.be>)
```

**Review by houssine78 — CHANGES_REQUESTED** [review]
very small changes

File: document_hosting/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
+
```
Comment (remytms) [inline document_hosting/__manifest__.py]: Remove this empty line at the beginning of the document. :)

File: document_hosting/__manifest__.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+# Copyright 2018 Rémy Taymans <remytaymans@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Document Hosting',
+
+    'summary': """
+    Manage the documents of your cooperative.
```
Comment (remytms) [inline document_hosting/__manifest__.py]: Suggestion: "Manage documents that can be published on website with `document_hosting_website`."

File: document_hosting/models/document.py [hunk 8]
```diff
@@ -0,0 +1,80 @@
+
```
Comment (remytms) [inline document_hosting/models/document.py]: Remove empty line in the beginning of the document.

File: document_hosting/models/document.py [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+# Copyright 2018 Rémy Taymans <remytaymans@gmail.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
```
Comment (remytms) [inline document_hosting/models/document.py]: Change license as Houssine suggested.

File: document_hosting/models/document.py [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+
+from odoo import models, fields, api
```
Comment (remytms) [inline document_hosting/models/document.py]: Can you use `isort` to sort imports on models. :)

File: document_hosting/views/document_hosting_views.xml [hunk 11]
```diff
@@ -0,0 +1,126 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+     Copyright 2018 Rémy Taymans <remytaymans@gmail.com>
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+-->
```
Comment (remytms) [inline document_hosting/views/document_hosting_views.xml]: Change license as Houssine suggested.

File: document_hosting/views/document_hosting_views.xml [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+<odoo>
+  <data>
```
Comment (remytms) [inline document_hosting/views/document_hosting_views.xml]: Remove `data` tag. Not needed.

**Review by remytms — CHANGES_REQUESTED** [review]
This is my Xmas wish list:
- Use `isort` on all python file except `__init__.py` ones.
- Paint code in black with `black -l 79 <module_name>`
- Change copyright holder everywhere as Houssine suggested.

Thanks a lot ! :)

**Comment by elouan-lb** [comment]
Thank you for the review @houssine78 & @remytms ! Changes were made, happy Christmas ! Migration and renaming of `easy_my_coop_website_document` into `document_hosting_website` is still needed.

**Review by houssine78 — APPROVED** [review]

**Review by remytms — APPROVED** [review]

**Comment by elouan-lb** [comment]
Added `easy_my_coop_website_document` as well. Documents display is configurable. See https://github.com/coopiteasy/vertical-cooperative/pull/58

File: document_hosting/__manifest__.py [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+    "name": "Document Hosting",
+    "summary": """
+    Manage documents that can be published on website with ??.
+    """,
+    "description": """
+    """,
+    "author": "Coop IT Easy SCRLfs",
+    "license": "AGPL-3",
+    "version": "12.0.1.0.0",
+    "website": "https://github.com/coopiteasy/addons",
```
Comment (houssine78) [inline document_hosting/__manifest__.py]: best to put www.coopiteasy.be instead

**Review by houssine78 — APPROVED** [review]

File: document_hosting/__manifest__.py (hunk 13 above)
Comment (elouan-lb) [reply document_hosting/__manifest__.py]: Done !

**Pull request merged**